src/AeroSharp/DataAccess/MapAccess/IMap.cs

```diff
@@ -19,28 +20,41 @@ public interface IMap<TKey, TValue>
     /// <param name="value"> The value to add. </param>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
     /// <returns> A task that represents the asynchronous operation of adding the entry. </returns>
-    Task PutAsync(TKey key, TValue value, CancellationToken cancellationToken);
+    Task PutAsync(TKey key, TValue value, CancellationToken cancellationToken, params CTX[] context);
```

Comment:
Couple of thoughts here:
- Not sure we should expose `CTX` as a type, since it's from a dependency.
- Not sure we should be using `params` here. Why not just an array?
- CancellationToken should go at the end.